### PR TITLE
[MOS-000] Added tool for translations verification

### PR DESCRIPTION
**Pull request by SP2FET**
Added script for verifying if the translation json:
- has duplicates
- has empty entries
- has missing entries

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [x] Has documentation updated
- [ ] Has changelog entry added

### CODE REVIEW

File: tools/verify_translations.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+# iterate to get all possible keys and check for key duplicates
+    for file in dir_list:
+        file_path = path.join(json_path, file)
+        check_duplicates(file_path)
+
+        with open(file_path) as json_file:
+            json_data = json.load(json_file)
+            check_empty_entry(file, json_data)
+
+            json_keys.append(set([key for key in json_data]))
```
Comment (pawelszalw) [inline tools/verify_translations.py:55]: List comprehension seems to be redundant in this place. This should work same:
`json_keys.append(set(json_data))`

File: tools/verify_translations.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    all_keys = set.union(*json_keys)
+
+# iterate to find missing keys
+    for file in dir_list:
+        file_path = path.join(json_path, file)
+        with open(file_path) as json_file:
+            json_data = json.load(json_file)
+            missing_keys_in_file = all_keys.difference(set([key for key in json_data]))
```
Comment (pawelszalw) [inline tools/verify_translations.py:64]: List comprehension seems to be redundant in this place. This should work same:
`missing_keys_in_file = all_keys.difference(set(json_data))`

File: tools/verify_translations.py (hunk 1 above)
Comment (SP2FET) [reply tools/verify_translations.py:55]: fixed

File: tools/verify_translations.py (hunk 2 above)
Comment (SP2FET) [reply tools/verify_translations.py:64]: fixed

File: tools/verify_translations.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        try:
+            _ = json.load(json_file, object_pairs_hook=validate_data)
+        except ValueError as e:
+            dup_list = str(e).split(',')
+            print(f"[{path.basename(file_path)}]: duplicate {len(dup_list)}: {dup_list}")
+
+
+def check_empty_entry(file_name: str, json_data: json):
+    empty_entries = [entry for entry in json_data if len(str(json_data[entry])) == 0]
+    if len(empty_entries) != 0:
```
[the remark this answers is not in the document]
Comment (SP2FET) [reply tools/verify_translations.py:38]: fixed

**Review by pawelszalw — APPROVED** [review]

**Review by lukasmas — APPROVED** [review]

**Review by dawidwojtas-mudita — APPROVED** [review]

File: tools/verify_translations.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    return ret
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        prog='verify_translations',
+        description='Script for checking the inconsistency of lang jsons',
+        formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument('-p', '--path', help="path to the json files", required=True)
+    parser.add_argument('--fix', action='store_true', help=textwrap.dedent('''\
```
Comment (swichu91) [inline tools/verify_translations.py:129]: How about adding optional argument(file path/name) to store the result of 'fix' operation?

File: tools/verify_translations.py [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    # data
+    return dict(list_of_pairs)
+
+
+def perform_on_files_from_path(json_path: path, operation):
+    dir_list = os.listdir(json_path)
+    ret = 0
+
+    for file in dir_list:
+        file_path = path.join(json_path, file)
```
Comment (swichu91) [inline tools/verify_translations.py:40]: If you import `from pathlib import Path` and use `Path` you would be able to replace `path.join` with simple `+ operator`.

**Review by swichu91 — APPROVED** [review]

File: tools/verify_translations.py (hunk 4 above)
Comment (SP2FET) [reply tools/verify_translations.py:129]: good idea, I'll add this one to prevent accidental overwriting the files

Comment (SP2FET) [reply tools/verify_translations.py:129]: added

File: tools/verify_translations.py (hunk 5 above)
Comment (SP2FET) [reply tools/verify_translations.py:40]: it works, but I wanted to leave it in a old-fashioned way :P

**Pull request merged**